File: Front/TS/tests/unit/implems/MockCompleteBetPresenter.ts
import { CompleteResponse, type ICompleteBetOutputPort } from "../../../src/domain/features/CompleteBetHandler";

export class MockCompleteBetPresenter implements ICompleteBetOutputPort {
    
    ProofIsRequired(): void {
        this.IsCompleted = false;
        this.ErrorMessage = "proof is required"
    }
    Success(response: CompleteResponse): void {
        this.IsCompleted = true;
    }
    IsCompleted: boolean;
    ErrorMessage?: string;
}

File: Front/.Net Maui/BetFriends.Domain.UnitTests/Implems/MockCompleteBetPresenter.cs
using BetFriends.Domain.Features.CompleteBet;

namespace BetFriends.Domain.UnitTests.Implems;

internal class MockCompleteBetPresenter : ICompleteBetOutputPort
{
    internal string Message { get; private set; }

    public void ProofIsMissing(string betId)
    {
        Message = $"proof is missing for bet {betId}";
    }

    public void Success(string betId, bool isSuccess)
    {
        Message = $"bet {betId} completed";
    }
}
